pkg/shp/reader_test.go

```diff
@@ -0,0 +1,221 @@
+package shp
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func assertPointsEqual(t *testing.T, a, b []float64, msgAndArgs ...interface{}) bool {
+
+	if !assert.True(t, len(a) == len(b), msgAndArgs...) {
+		return false
+	}
+	for k, v := range a {
+		if !assert.True(t, v == b[k], msgAndArgs...) {
+			return false
+		}
+	}
+	return true
+}
+
+func getShapesFromFile(prefix string, t *testing.T) (shapes []Shape) {
+	filename := prefix + ".shp"
+	ior, _ := os.Open(filename)
+	file, err := ReadFrom(ior)
+	assert.Nil(t, err, "Failed to open shapefile")
+
+	defer func(ior *os.File) {
+		err := ior.Close()
+		assert.Nil(t, err, "Failed to close shapefile")
+	}(ior)
```

Comment:
This is exactly the same code as below, since go's anonymous function automatically captures the variables:

```suggestion
	defer func() {
		err := ior.Close()
		assert.Nil(t, err, "Failed to close shapefile")
	}()
```